ci/tests/csvpump/test.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash  
+
+set -eao pipefail
+
+function usage {
+    local progname=$1
+    cat <<EOF
+Usage:
+
+    $progname  tag [gwtag]
+
+    tag   -  The tag for the pump image, that needs to be tested.
+    gwtag -  (Optional) The  tag for the GW image, that needs to be tested against the pump.
+             If not given, the default version would be used, which is $DEFAULT_GW_TAG
+
+Brings up the given tyk gw, and tyk pump and checks if the csv pump is
+working properly.
+Requires docker compose.
+
+EOF
+    exit 1
+}
+
+function info {
+    local text=$1
+    echo -e "\033[0;36m $text \033[0m"
+}
+
+function warn {
+    local text=$1
+    echo -e "\033[0;31m $text \033[0m"
+}
+
+# The default gateway image that will be pulled and checked against in case
+# no explicit gateway image tag is given.
+DEFAULT_GW_TAG=v4.0.1
+
+[[ -z $1 ]] && usage "$0"
+export tag=$1
+export gwtag=$2
+
+if [[ -z $2 ]]; then
+    warn "No explicit gateway tag given - Will use $DEFAULT_GW_TAG"
+    gwtag=$DEFAULT_GW_TAG
+fi
+
+compose='docker-compose'
+# use the compose client plugin if v2
+[[ $(docker version --format='{{ .Client.Version }}') =~ 20.10 ]] && compose='docker compose'
+
+#create the tmp directory to hold pump data.
+if [[ -e /tmp/pump-data ]]; then
+    rm -rf /tmp/pump-data/*.csv
+else
+    mkdir /tmp/pump-data
+fi
+
+trap '$compose down' EXIT
+
+$compose up -d
+
+GWBASE="http://localhost:8080"
+
+
+curlf() {
+    curl --header 'content-type:application/json' -s --show-error "$@"
+}
+
+cleanup() {
+    rm -f /tmp/pump-data/*.csv
+}
+
+check_gw_status() {
+    info "Checking Tyk GW status..."
+    status=$(curlf "${GWBASE}/hello" | jq -r '.status')
+    if [ "$status" != "pass" ]
+    then
+        return 1
+    fi
+    redis_status=$(curlf "${GWBASE}/hello" | jq -r '.details.redis.status')
+    if [ "$redis_status" != "pass" ]
+    then
+        return 1
+    fi
+    return 0
+}
```

Comment:
This is good, what do you think about moving this into the docker-compose like this?
    healthcheck:
      test: curl -s --fail http://localhost:8080/hello | grep -o '.*status":"pass".*details":{.*redis":{"status":"pass"'
      interval: 5s
      retries: 10
      start_period: 4s
      timeout: 10s